spec/lib/sequent/core/workflow_spec.rb

```diff
@@ -14,14 +16,24 @@ class SendWelcomeEmail < Sequent::Core::BaseCommand
 
   class RegistrationWorkflow < Sequent::Core::Workflow
     on UserWasRegistered do |e|
-      execute_commands SendWelcomeEmail.new(email: e.email)
+      execute_commands CreateNotification.new(aggregate_id: e.aggregate_id)
+
+      after_commit do
+        execute_commands SendWelcomeEmail.new(email: e.email)
```

Comment:
The `execute_commands` will start a new transaction. Is that possible in an `after_commit` hook?

Reply:
As long as it doesn't cause a loop, i.e. trigger another `UserWasRegistered`, this will be fine.

Reply:
Yes, `execute_commands` will start a new transaction and yes, it is possible in an `after_commit`. I just tried it. @stephanvd is right but a loop can be created even without the `after_commit`, right?

Reply:
Yes indeed, so we do not have to do anything about that.